lib/explorer/data_frame.ex

```diff
@@ -1525,6 +1525,40 @@ defmodule Explorer.DataFrame do
     Shared.apply_impl(df, :to_rows, [opts[:atom_keys]])
   end
 
+  @doc """
+  Converts a dataframe to a stream of maps (rows).
+
+  > #### Warning {: .warning}
+  >
+  > This is an expensive operation since data is stored in a columnar format.
+  > Prefer to use the operations in this module rather than the ones in `Enum`
+  > whenever possible, as this module is optimized for large series.
+
+  ## Options
+
+    * `:atom_keys` - Configure if the resultant maps should have atom keys. (default: `false`)
+    * `:buffer_size` - The number of rows that are buffered internally while iterating over the data. (default: `1000`)
```

Comment:
Can we call it chunk_size? I don't think buffer size is correct because buffer means it is somehow being accumulated somewhere (in disk or in memory) but we are really just traversing it using a chunk.